src/dataflow-types/src/types.rs

```diff
@@ -769,12 +769,12 @@ pub mod sources {
     #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
     pub enum SourceEnvelope {
         /// The most trivial version is `None`, which typically produces triples where the diff
-        /// is ALWAYS `1`
+        /// is `1`. However, some sources are able to produce values with more exotic diff's,
+        /// such as the posgres source. Currently, this is the only variant usable with
+        /// those sources.
```

Comment:
Can you say a bit about why this change? Up until now `Envelope::None` had meant "append", and .. it now means something else.

Reply:
This was actually already broken: `SimpleSource`'s could already retract/delete rows, and they used `SourceEnvelope::None`

Reply:
Pushing the bug further down seems like maybe the wrong outcome?

Reply:
@frankmcsherry I am not sure it is pushing it down, I fixed the `optimize_dataflow_monotonic`, which as far as I can tell, was the only erroneous assumption. that said, I can definitely add an envelope and clean up the sql parsing and type-checking in a later pr!

Reply:
Let me restate: historically `None` for external sources meant "append only" and was confirmed by the sources/sinks folks to mean that. I agree it was never documented. Asking `transform` (not a storage-y crate) to track which combinations of envelopes are "append only" seems like an abstraction violation. STORAGE should provide a `fn is_append_only` method on some sort of source identifier, so that the `transform` crate can rely on it rather than the shifting meanings of this enum.

Reply:
@frankmcsherry Oh see! Yes I will change it to that!

Reply:
@guswynn @aljoscha has added a new Envelope type called `DifferentialRow` in order to discern between append-only and collection-like things. Should use that for postgres sources?

Reply:
@petrosagg I think so, pg cant have a `KeyEnvelope` other than `None` right?

Reply:
https://github.com/MaterializeInc/materialize/issues/12413